Expecto/Test.fs

```diff
@@ -141,4 +141,5 @@ module Test =
     | Async test -> timeoutAsync test |> Async
     | AsyncFsCheck (testConfig, stressConfig, test) ->
       AsyncFsCheck (testConfig, stressConfig, test >> timeoutAsync)
+    | AsyncTask test -> timeoutAsync (test |> Async.AwaitTask) |> Async
```

Comment:
Might we as well convert to async right away (i.e. in `testCaseTask`) instead of adding a new `TestCode` union case?
Is there a reason you chose the extra union case?

Reply:
> Is there a reason you chose the extra union case?

I thought that it would be better to explicitly say the test is a task than to pass it as a converted `Async<'T>`, but now I think this is not useful.